### PR TITLE
pass through host header

### PR DESCRIPTION
**Pull request by johannakate**
This needs work on the test

### CODE REVIEW

**Comment by johannakate** [comment]
@martijnwalraven @zionts this test isn't really where I want it to be but will take a look tomorrow

File: test/koa.js [hunk 1]
```diff
@@ -93,6 +111,31 @@ describe('koa middleware', () => {
           resolve();
         });
       })
-    })
+    });
+
+    it.only('passes the request host header', () => {
```
Comment (zionts) [inline test/koa.js]: Remove only and also maybe just remove this test if it's failing. The work is saved in git so it won't be lost.

File: test/schema.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ exports.verifyEndpointSuccess = (url, hasTracing) => {
       json: true,
       body: {'query': '{ hello }'}
     }, (err, response, body) => {
+      console.log('response: ', response.statusCode, response.data);
```
Comment (zionts) [inline test/schema.js]: remove console.log

File: src/middleware.ts [hunk 3]
```diff
@@ -62,22 +62,23 @@ export function makeKoaMiddleware(params: MiddlewareParams) {
...
-                    else {
-                        ctx.response.status = response.statusCode;
-                        ctx.response.set(JSON.parse(JSON.stringify(response.headers)));
-                        ctx.response.body = body;
-                        resolve();
-                    }
-                }));
-            });
+            ctx.req.pipe(request(params.uri + ctx.originalUrl,
+                { headers: { 'host': ctx.req.headers.host } },
```
Comment (zionts) [inline src/middleware.ts]: Also it looks like you can just do `ctx.set('host', ctx.req.headers.host)` according to https://github.com/koajs/koa/blob/master/docs/guide.md#writing-middleware

File: src/middleware.ts [hunk 4]
```diff
@@ -57,27 +57,26 @@ export function makeConnectMiddleware(params: MiddlewareParams) {
...
-              ctx.response.set(JSON.parse(JSON.stringify(response.headers)));
-              ctx.response.body = body;
-              resolve();
-            }
-          }),
-        );
-      });
-  };
+    return function (ctx: Context, next: () => Promise<any>) {
+        if (!params.uri || ctx.path !== params.endpoint) return next();
```
Comment (zionts) [inline src/middleware.ts]: Can you fix tabbing? lint tests will fail :/

**Comment by zionts** [comment]
@johannakate I have the formatting set up in my vscode to make sure it complies to the linting test, so I'll push the linting changes over your branch if that's okay :) Also on Monday I can help you set up vscode with the linting plugin.

**Pull request merged**